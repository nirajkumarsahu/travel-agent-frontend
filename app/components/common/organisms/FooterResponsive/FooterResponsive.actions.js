import {
  FETCH_FOOTER_DATA,
  FOOTER_DATA_SUCCESS,
  FOOTER_LOAD_FAILURE
} from "./FooterResponsive.constants";

export const FooterLoadSuccess = data => ({
  type: FOOTER_DATA_SUCCESS,
  data
});

export const FooterLoadFailure = data => ({
  type: FOOTER_LOAD_FAILURE,
  data
});

export const fetchFooterData = () => ({
  type: FETCH_FOOTER_DATA
});
