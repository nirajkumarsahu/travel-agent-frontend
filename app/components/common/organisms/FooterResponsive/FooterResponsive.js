import React, { useState } from "react";
import PropTypes from "prop-types";
import styledHOC from "lib/styledHOC";
import FooterWhiteBox from "components/common/molecules/FooterWhiteBox";
import FooterSiteMap from "components/common/molecules/FooterSiteMap/FooterSiteMap";
import FooterImages from "components/common/molecules/FooterImages";
import FooterPayment from "components/common/molecules/FooterPayment";
import FooterHotelsLinkResponsive from "components/common/molecules/FooterHotelsLinkResponsive";
import Heading from "components/common/atoms/Heading";
import styles from "./FooterResponsive.style";

const mockData = [
  {
    heading: "Fabhotels",
    Anchorlist: [
      {
        url: "/about-us",
        anchorText: "About Us"
      },
      {
        url: "/careers",
        anchorText: "Careers"
      },
      {
        url: "/press",
        anchorText: "Press"
      },
      {
        url: "/blog",
        anchorText: "Travel Blog"
      }
    ]
  },

  {
    heading: "Partner with us",
    Anchorlist: [
      {
        url: "/franchise-enquiry",
        anchorText: "Be our Franchisee"
      },
      {
        url: "/travel-agent",
        anchorText: "Travel Agent"
      },
      {
        url: "/corporate-enquiry",
        anchorText: "Corporate Enquiries"
      }
    ]
  },
  {
    heading: "Policies",
    Anchorlist: [
      {
        url: "/terms-conditions",
        anchorText: "Terms & Conditions"
      },
      {
        url: "/privacy-policy",
        anchorText: "Privacy Policy"
      },
      {
        url: "/cancellation-refunds-policy",
        anchorText: "Cancellation Policy"
      },
      {
        url: "/faqs",
        anchorText: "FAQs"
      }
    ]
  }
];

const mockDataSignup = [
  {
    heading: "Partner with us",
    Anchorlist: [
      {
        url: "/franchise-enquiry",
        anchorText: "Be our Franchisee"
      },
      {
        url: "/travel-agent",
        anchorText: "Travel Agent"
      },
      {
        url: "/corporate-enquiry",
        anchorText: "Corporate Enquiries"
      }
    ]
  },
  {
    heading: "Policies",
    Anchorlist: [
      {
        url: "/terms-conditions",
        anchorText: "Terms & Conditions"
      },
      {
        url: "/privacy-policy",
        anchorText: "Privacy Policy"
      },
      {
        url: "/cancellation-refunds-policy",
        anchorText: "Cancellation Policy"
      }
    ]
  }
];

const FooterResponsive = props => {
  // eslint-disable-next-line react/prop-types
  const {
    className,
    footerData = {},
    getSVG,
    user,
    isMobile,
    signUpFooter,
    svgLoaded
  } = props;
  const { footer = {} } = footerData;
  const [displayList, setListFlag] = useState(false);
  const ArrowWhite = getSVG("arrowWhite");
  const ShieldLine = getSVG("shieldLine");
  const PhoneLine = getSVG("phoneLine");
  const LocationLine = getSVG("locationLine");
  let tempData;
  if (signUpFooter) {
    tempData = mockDataSignup;
  } else {
    tempData = mockData;
  }
  const SiteMapRepeater = tempData.map(siteMap => (
    <FooterSiteMap
      anchorList={siteMap.Anchorlist}
      headingText={siteMap.heading}
    />
  ));
  return (
    <footer className={className}>
      {/* eslint-disable-next-line */}
      <div
        className={`visible-phone book-hotels ${
          displayList ? "footer-open" : ""
        }`}
        onClick={() => setListFlag(!displayList)}
      >
        Book our hotels in all these cities
        <span className="arrowIcon">
          <ArrowWhite />
        </span>
      </div>
      <FooterWhiteBox className="white-box" />
      {(displayList || !isMobile) && (
        <div className="container footer-slide">
          <div className="link-wrapper flex">
            <div className="sitemap">
              <div className="flex link-sec">{SiteMapRepeater}</div>
              <FooterPayment getSVG={getSVG} />
            </div>
            {!signUpFooter ? (
              <FooterImages svgLoaded={svgLoaded} user={user} getSVG={getSVG} />
            ) : (
              <div className="footer-option">
                <div className="about-us">
                  <Heading type="h4" tag="h3">
                    About Fabhotels
                  </Heading>
                  <ul>
                    <li>
                      <ShieldLine />
                      2019 Casa2 Stays Pvt. Ltd. All rights reserved. Â©
                    </li>
                  </ul>
                  <ul>
                    <li>
                      <PhoneLine />
                      Plot No 53 and 54, Phase 4, Sub.Major Laxmi Chand Road,
                      Udyog Vihar, Gurugram, Haryana 122022
                    </li>
                  </ul>
                  <ul>
                    <li>
                      <LocationLine />
                      +1 (234) 567 8900
                    </li>
                  </ul>
                </div>
              </div>
            )}
          </div>
          <FooterHotelsLinkResponsive
            locationHeading="Book our hotels in all these cities"
            descHeading="FabHotels is Best Reviewed Hotel Chain in India"
            compData={footer}
          />
        </div>
      )}
      <div className="container">
        <div className="copyright">
          <p>&#9400; 2019 Casa2 Stays Pvt. Ltd. All rights reserved.</p>
        </div>
      </div>
    </footer>
  );
};

FooterResponsive.propTypes = {
  className: PropTypes.string.isRequired,
  getSVG: PropTypes.func.isRequired,
  footerData: PropTypes.string.isRequired,
  signUpFooter: PropTypes.bool,
  user: PropTypes.instanceOf(Object).isRequired,
  isMobile: PropTypes.bool.isRequired,
  svgLoaded: PropTypes.bool.isRequired
};

FooterResponsive.defaultProps = {
  signUpFooter: false
};

export { FooterResponsive as FooterVanilla };
export default styledHOC(FooterResponsive, styles);
