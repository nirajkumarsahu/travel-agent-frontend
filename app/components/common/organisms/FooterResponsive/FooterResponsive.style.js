import { css } from "styled-components";

export default css`
  padding: 75px 0 10px;
  position: relative;
  background-image: linear-gradient(144deg, #192454, #525e93);
  .white-box {
    display: none;
  }
  .link-wrapper {
    flex-wrap: wrap;
    padding-bottom: 42px;
    margin-bottom: 41px;
    position: relative;
    &::after {
      content: "";
      position: absolute;
      left: 0;
      bottom: 0;
      width: 100%;
      height: 2px;
      background-color: ${props => props.theme.color_footerLine};
      opacity: 0.15;
    }
  }
  .white-box {
    position: absolute;
    bottom: 100%;
    left: 50%;
    transform: translate(-50%, 50%);
    -webkit-transform: translate(-50%, 50%);
  }
  .sitemap {
    width: 49%;
  }
  .link-sec {
    margin: 0 0 40px;
  }

  .book-hotels {
    font-size: ${props => props.theme.fs_small}px;
    color: ${props => props.theme.color_White};
    margin: 0 0 10px;
    line-height: 1.35;
    padding: 0 12px 10px;
    font-weight: ${props => props.theme.fw_SemiBold};
    position: relative;
    border-bottom: 1px solid ${props => props.theme.color_White};

    .arrowIcon {
      position: absolute;
      right: 9px;
      top: 0;
      left: auto;
      transform: rotate(90deg);
      transition: all ease 0.3s;

      svg {
        width: 11px;
        height: 11px;
      }
    }
  }

  .book-hotels.footer-open {
    .arrowIcon {
      transform: rotate(-90deg);
    }
  }

  .copyright {
    padding: 20px 0 0;
    p {
      text-align: center;
      margin: 0;
      color: ${props => props.theme.color_White};
      font-size: ${props => props.theme.fs_xsmall}px;
    }
  }

  .footer-option {
    width: 50%;
    border-left: 1px solid rgba(181, 181, 181, 0.15);
    display: flex;
    flex-wrap: wrap;
  }

  .footer-option:last-child {
    border-right: none;
  }

  .about-us {
    width: 100%;
    padding-left: 40px;
    color: ${props => props.theme.color_White};
    h3 {
      color: ${props => props.theme.color_White};
    }
    ul {
      padding: 0;
      margin: 12px 0;
    }
    li {
      display: flex;
      font-size: ${props => props.theme.fs_small}px;
      line-height: 20px;
      opacity: 0.7;
    }
  }

  .phoneLine_svg__phoneLine-icon,
  .locationLine_svg__locationLine-icon,
  .shieldLine_svg__shieldLine-icon {
    width: 16px;
    height: 16px;
    vertical-align: middle;
    margin: 2px 16px 0 0;
  }

  .phoneLine_svg__phoneLine-fill,
  .locationLine_svg__locationLine-fill,
  .shieldLine_svg__shieldLine-fill {
    fill: ${props => props.theme.color_White};
  }

  @media ${props => props.theme.mediaQuery.medium} {
    padding: 144px 0 40px;
  }
`;
