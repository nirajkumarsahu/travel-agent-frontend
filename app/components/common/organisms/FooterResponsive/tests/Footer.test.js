import React from "react";
import { shallow } from "enzyme";
import { FooterVanilla } from "../FooterResponsive";

describe("Footer Test Cases", () => {
  const props = {
    className: "testClass"
  };
  let output = "";

  beforeEach(() => {
    output = shallow(<FooterVanilla {...props} />);
  });

  test("should render correctly", () => {
    expect(output).toMatchSnapshot();
  });
});
