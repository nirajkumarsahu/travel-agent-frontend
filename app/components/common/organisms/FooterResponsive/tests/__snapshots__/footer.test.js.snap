// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Test Cases should render correctly 1`] = `
<footer
  className="testClass"
>
  <Link
    href="/"
  >
    <a>
      Home
    </a>
  </Link>
  <Link
    href="/about"
  >
    <a>
      About
    </a>
  </Link>
</footer>
`;
